floss/language/go/extract.py

```diff
@@ -0,0 +1,511 @@
+    0048E660  09 2A 5B 5D 75 69 6E 74  36 34 00 09 2A 63 68 61  .*[]uint64..*cha
+    0048E670  6E 20 69 6E 74 01 09 41  6E 6F 6E 79 6D 6F 75 73  n int..Anonymous
+    0048E680  01 09 43 61 6C 6C 53 6C  69 63 65 01 09 43 6C 65  ..CallSlice..Cle
+    0048E690  61 72 42 75 66 73 01 09  43 6F 6E 6E 65 63 74 45  arBufs..ConnectE
+    0048E6A0  78 01 09 46 74 72 75 6E  63 61 74 65 01 09 49 6E  x..Ftruncate..In
+    0048E6B0  74 65 72 66 61 63 65 01  09 4E 75 6D 4D 65 74 68  terface..NumMeth
+    0048E6C0  6F 64 01 09 50 72 65 63  69 73 69 6F 6E 01 09 52  od..Precision..R
+    """
+
+    # The blob start like 00 00 <data> and ends like this:
+    """
+    0000000000492A40  6C 65 63 74 2E 53 74 72  75 63 74 46 69 65 6C 64  lect.StructField
+    0000000000492A50  2C 20 62 6F 6F 6C 29 00  00 00 00 00 00 00 00 00  , bool).........
+    0000000000492A60  08 00 00 00 00 00 00 00  08 00 00 00 00 00 00 00  ................
+    """
+
+    # Get the blob
+    blob_end_pattern = re.compile(b"\x00{8}.\x00{7}.\x00{7}")
+    blob_end = re.search(blob_end_pattern, section_data[2:])
+    if not blob_end:
+        return []
+    blob = section_data[2 : blob_end.start()]
+
+    reflection_strings = list()
+
+    # Extract strings from the blob
+    blob_pattern = re.compile(b"(\x00|\x01|\x02|\x03)(?P<string_length>.)", re.DOTALL)
+    for m in blob_pattern.finditer(blob):
+        if m.group("string_length") != b"\x00":
+            try:
+                addr = pe.get_offset_from_rva(m.start() + section_va + 4)
+                binary_string = blob[m.end() : m.end() + m.group("string_length")[0]]
+                try:
+                    string = StaticString.from_utf8(binary_string, addr, min_length)
+                    reflection_strings.append(string)
+                except ValueError:
+                    pass
+            except pefile.PEFormatError:
+                pass
+    return reflection_strings
+
+
+def extract_string_blob2(pe: pefile.PE, section_data, section_va, min_length) -> List[StaticString]:
+    # Extract string blob in .rdata section that starts with "go:buildid" or "go.buildid"
+    """
+    67 6F 3A 62 75 69 6C 64  69 64 00 69 6E 74 65 72  go:buildid.inter
+    6E 61 6C 2F 63 70 75 2E  49 6E 69 74 69 61 6C 69  nal/cpu.Initiali
+    7A 65 00 69 6E 74 65 72  6E 61 6C 2F 63 70 75 2E  ze.internal/cpu.
+    70 72 6F 63 65 73 73 4F  70 74 69 6F 6E 73 00 69  processOptions.i
+    6E 74 65 72 6E 61 6C 2F  63 70 75 2E 69 6E 64 65  nternal/cpu.inde
+    78 42 79 74 65 00 69 6E  74 65 72 6E 61 6C 2F 63  xByte.internal/c
+    """
+
+    # Get the blob
+    blob_start_pattern = re.compile(b"go(\.|:)buildid\x00")
+    blob_start = re.search(blob_start_pattern, section_data)
+    if not blob_start:
+        return []
+    blob = section_data[blob_start.start() :]
+    blob_end_pattern = re.compile(b"\x00{2}")
+    blob_end = re.search(blob_end_pattern, blob)
+    if not blob_end:
+        return []
+    blob = blob[: blob_end.start()]
```

Comment:
```suggestion
    blob = section_data.partition(b"buildid\x00")[2].partition(b"\x00\x00")[0]
```

*slightly* less expressive but simpler code. thoughts?